src/headers/frame_header.rs

```diff
@@ -0,0 +1,494 @@
+#![allow(clippy::excessive_precision)]
+
+use crate::{bit_reader::BitReader, error::Error, headers::encodings::*};
+
+use jxl_headers_derive::UnconditionalCoder;
+use num_derive::FromPrimitive;
+
+#[derive(UnconditionalCoder, Copy, Clone, PartialEq, Debug, FromPrimitive)]
+enum FrameType {
+    RegularFrame = 0,
+    LFFrame = 1,
+    ReferenceOnly = 2,
+    SkipProgressive = 3,
+}
+
+#[derive(UnconditionalCoder, Copy, Clone, PartialEq, Debug, FromPrimitive)]
+enum Encoding {
+    VarDCT = 0,
+    Modular = 1,
+}
+
+#[derive(UnconditionalCoder, Debug, PartialEq)]
+struct Passes {
+    #[coder(u2S(1, 2, 3, Bits(3) + 4))]
+    #[default(1)]
+    num_passes: u32,
+
+    #[coder(u2S(0, 1, 2, Bits(1) + 3))]
+    #[default(0)]
+    #[condition(num_passes != 1)]
+    num_ds: u32,
+
+    #[size_coder(explicit(num_passes - 1))]
+    #[coder(Bits(2))]
+    #[default(Vec::new())]
```

Comment:
IIRC this is implicit, no? If so, I'd probably rather not mention it (it likely looks ugly in the generated `.tex` files - IDK if you tried `make_all_tex.sh`)

Reply:
I should've used better words than implicit / explicit.
My intention was to make it easy to distinguish between expressions like `num_passes` and `Bits(3) + 4` inside the size_coder. Although looking at the draft, size_coders only use `u2S`, so we don't need to care about it.

Reply:
Implicit/explicit is OK (but maybe I'd go with computed/signaled), but I was referring to the default value :)

Reply:
Oh, got it, I needed to include those in order for `Passes` to implement `Default`.

Reply:
Sounds good, thanks!